data/transactions/logic/crypto_test.go

```diff
@@ -264,6 +297,44 @@ func TestEd25519VerifyBare(t *testing.T) {
 	}
 }
 
+func TestFalconVerify(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	var s crypto.FalconSeed
+	fs, err := crypto.GenerateFalconSigner(s)
+	require.NoError(t, err)
+
+	msg := "62fdfc072182654f163f5f0f9a621d729566c74d0aa413bf009c9800418c19cd"
+	data, err := hex.DecodeString(msg)
+	require.NoError(t, err)
+
+	for v := uint64(11); v <= AssemblerMaxVersion; v++ {
+		t.Run(fmt.Sprintf("v=%d", v), func(t *testing.T) {
+			ops := testProg(t, fmt.Sprintf(`arg 0; arg 1; byte 0x%s; falcon_verify`, hex.EncodeToString(fs.PublicKey[:])), v)
+			require.NoError(t, err)
+			sig, err := fs.SignBytes(data)
+			require.NoError(t, err)
+
+			var txn transactions.SignedTxn
+			txn.Lsig.Logic = ops.Program
+			txn.Lsig.Args = [][]byte{data[:], sig[:]}
+			testLogicBytes(t, ops.Program, defaultSigParams(txn))
+
+			// short sig will fail
+			txn.Lsig.Args[1] = sig[1:]
+			testLogicBytes(t, ops.Program, defaultSigParams(txn), "verify failed")
```

Comment:
This should reject as well, right? So why does it expect the error "verify failed"?

Reply:
Yes, that you for catching that. I am incorrectly propagating the `err` from the verify for the opcode, so all failed verifies will panic.  Shows I need to have passing tests in which the verify fails.